hmda-analytics/src/main/scala/hmda/analytics/HmdaAnalyticsApp.scala

```diff
@@ -347,27 +328,27 @@ object HmdaAnalyticsApp extends App with TransmittalSheetComponent with LarCompo
         .mapAsync(1) { lar =>
           for {
             delete <- submissionId.period match {
-              case Period(2018, None) => larRepository2018.deleteByLei(lar.larIdentifier.LEI)
-              case Period(2019, None) => larRepository2019.deleteByLei(lar.larIdentifier.LEI)
-              case Period(2020, Some("Q1")) => larRepository2020Q1.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2020, Some("Q2")) => larRepository2020Q2.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2020, Some("Q3")) => larRepository2020Q3.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2020, None) => larRepository2020.deleteByLei(lar.larIdentifier.LEI)
-              case Period(2021, None) => larRepository2021.deleteByLei(lar.larIdentifier.LEI)
-              case Period(2022, None) => larRepository2022.deleteByLei(lar.larIdentifier.LEI)
-              case Period(2023, None) => larRepository2023.deleteByLei(lar.larIdentifier.LEI)
-              case Period(2021, Some("Q1")) => larRepository2021Q1.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2021, Some("Q2")) => larRepository2021Q2.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2021, Some("Q3")) => larRepository2021Q3.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2022, Some("Q1")) => larRepository2022Q1.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2022, Some("Q2")) => larRepository2022Q2.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2022, Some("Q3")) => larRepository2022Q3.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2023, Some("Q1")) => larRepository2023Q1.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2023, Some("Q2")) => larRepository2023Q2.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2023, Some("Q3")) => larRepository2023Q3.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2024, Some("Q1")) => larRepository2024Q1.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2024, Some("Q2")) => larRepository2024Q2.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
-              case Period(2024, Some("Q3")) => larRepository2024Q3.deletebyLeiAndQuarter(lar.larIdentifier.LEI)
+              case Period(2018, None) => YearlyLarRepositoryWrapper("2018").getLarRepository.deleteByLei(lar.larIdentifier.LEI)
```

Comment:
Does this work? I just noticed above that YearlyLarRepositoryWrapper doesn't have a deleteByLei method. Seems to be unique in that way.

Reply:
You're correct it doesn't have one - it grabs a repo object and calls that ones deleteByLei. It does work since its doing that. I chose not to reimplement the lar repo functions in the wrapper only because the lar repo is such a small class. No other reason, good catch!